src/classes/VOL_SharedCode.cls

```diff
@@ -98,9 +128,9 @@ global with sharing class VOL_SharedCode {
 
         for (Volunteer_Shift__c vs : listVolunteerJobs[0].Volunteer_Job_Slots__r) {
             SelectOption so = new SelectOption(vs.id,
-                (useDateTimeFixup ? vs.System_Note__c : vs.Start_Date_Time__c.format()) +
+                canReadDate ? (useDateTimeFixup ? vs.System_Note__c : vs.Start_Date_Time__c.format()) : '' +
```

Comment:
@scottwarren-sfdo, I waffled on what to do here if the user does not have access to the date field and / or the number of vol needed field but do have object access. The shift name is not always included in the label, the date and conditionally the number of vol still needed make up the label (this is also used on personal site report hours). My gut reaction was just to return it anyway but technically they shouldn't see the data so I went with my 2nd thought here and just display the data if they have access. However, this could potentially lead to a list with no labels and ids only... Let me know your thoughts and or if you'd like to sync up on this.

Reply:
I agree with the position you took. If the user does not have access, we should not pass the data along. The resulting suboptimal UI is a result of a permissions misconfiguration.